x-pack/filebeat/input/entityanalytics/internal/collections/tree.go

```diff
@@ -62,10 +48,7 @@ func (t *Tree[T]) HasVertex(value T) bool {
 
 // AddEdge adds an edge (from, to).
 func (t *Tree[T]) AddEdge(from T, to ...T) {
```

Comment:
In this particular implementation and usage of it, that's correct. Originally, I did have vertexes without any edges, but never ended up using that for anything. I also originally had this outside the input as a generic package, but moved it inside since its usage became more opinionated. It'll be interesting to see if Okta or any other provider will need more functionality out of it, or not even use it at all.